From 767ffca1377150b7fb86ef1e58a01aea22096124 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Mon, 20 Jul 2020 14:10:04 -0500
Subject: [PATCH 21/23] board: imx8m/n beacon: Enable RAMDisk booting

The current boot scripts are only setup to load from eMMC, SD cards
or Network booting.

This script adds support for RAMdisk booting.  The default ramdisk
image is called rootfs.cpio.uboot, but it can be changed.

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 include/configs/imx8mm_beacon.h | 11 ++++++++++-
 include/configs/imx8mn_beacon.h | 11 ++++++++++-
 2 files changed, 20 insertions(+), 2 deletions(-)

diff --git a/include/configs/imx8mm_beacon.h b/include/configs/imx8mm_beacon.h
index 484119cf6e..165c09ebe7 100644
--- a/include/configs/imx8mm_beacon.h
+++ b/include/configs/imx8mm_beacon.h
@@ -119,9 +119,11 @@
 	JAILHOUSE_ENV \
 	"script=boot.scr\0" \
 	"image=Image\0" \
+	"ramdiskimage=rootfs.cpio.uboot\0" \
 	"console=ttymxc1,115200 earlycon=ec_imx6q,0x30890000,115200\0" \
 	"fdt_addr=0x43000000\0"			\
 	"fdt_high=0xffffffffffffffff\0"		\
+	"ramdisk_addr=0x44000000\0" \
 	"boot_fdt=try\0" \
 	"fdt_file=" CONFIG_DEFAULT_FDT_FILE "\0" \
 	"initrd_addr=0x43800000\0"		\
@@ -132,11 +134,14 @@
 	"mmcautodetect=yes\0" \
 	"mmcargs=setenv bootargs ${jh_clk} console=${console} " \
 	" root=PARTUUID=${uuid} rootwait rw ${mtdparts} ${optargs}\0" \
+	"ramargs=setenv bootargs ${jh_clk} console=${console} root=/dev/ram rw " \
+	" ${optargs}\0" \
 	"loadbootscript=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
 	"loadimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
 	"loadfdt=fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0" \
+	"loadramdisk=load mmc ${mmcdev} ${ramdisk_addr} ${ramdiskimage}\0"\
 	"mmcboot=echo Booting from mmc ...; " \
 		"run finduuid; run mmcargs; " \
 		"if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
@@ -167,7 +172,11 @@
 			"fi; " \
 		"else " \
 			"booti; " \
-		"fi;\0"
+		"fi;\0" \
+	"ramboot=echo Booting from RAMdisk...; "\
+		"run loadimage; run loadfdt; fdt addr $fdt_addr; "\
+		"run loadramdisk; run ramargs; " \
+		"booti ${loadaddr} ${ramdisk_addr} ${fdt_addr} ${optargs}\0"
 
 #define CONFIG_BOOTCOMMAND \
 	   "mmc dev ${mmcdev}; if mmc rescan; then " \
diff --git a/include/configs/imx8mn_beacon.h b/include/configs/imx8mn_beacon.h
index b31035c90b..28772477ce 100644
--- a/include/configs/imx8mn_beacon.h
+++ b/include/configs/imx8mn_beacon.h
@@ -85,9 +85,11 @@
 	JAILHOUSE_ENV \
 	"script=boot.scr\0" \
 	"image=Image\0" \
+	"ramdiskimage=rootfs.cpio.uboot\0" \
 	"console=ttymxc1,115200\0" \
 	"fdt_addr=0x43000000\0"			\
 	"fdt_high=0xffffffffffffffff\0"		\
+	"ramdisk_addr=0x44000000\0" \
 	"boot_fdt=try\0" \
 	"fdt_file=" CONFIG_DEFAULT_FDT_FILE "\0" \
 	"initrd_addr=0x43800000\0"		\
@@ -98,11 +100,14 @@
 	"mmcautodetect=yes\0" \
 	"mmcargs=setenv bootargs ${jh_clk} console=${console} " \
 	" root=PARTUUID=${uuid} rootwait rw ${mtdparts} ${optargs}\0" \
+	"ramargs=setenv bootargs ${jh_clk} console=${console} root=/dev/ram rw " \
+	" ${optargs}\0" \
 	"loadbootscript=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
 	"loadimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
 	"loadfdt=fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0" \
+	"loadramdisk=load mmc ${mmcdev} ${ramdisk_addr} ${ramdiskimage}\0"\
 	"mmcboot=echo Booting from mmc ...; " \
 		"run finduuid; run mmcargs; " \
 		"if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
@@ -133,7 +138,11 @@
 			"fi; " \
 		"else " \
 			"booti; " \
-		"fi;\0"
+		"fi;\0" \
+	"ramboot=echo Booting from RAMdisk...; "\
+		"run loadimage; run loadfdt; fdt addr $fdt_addr; "\
+		"run loadramdisk; run ramargs; " \
+		"booti ${loadaddr} ${ramdisk_addr} ${fdt_addr} ${optargs}\0"
 
 #define CONFIG_BOOTCOMMAND \
 	   "mmc dev ${mmcdev}; if mmc rescan; then " \
-- 
2.17.1

